datarepo-clienttests/src/main/java/scripts/testscripts/SnapshotScaleDelete.java

```diff
@@ -0,0 +1,124 @@
+import bio.terra.datarepo.client.ApiClient;
+import bio.terra.datarepo.model.DeleteResponseModel;
+import bio.terra.datarepo.model.JobModel;
+import bio.terra.datarepo.model.SnapshotModel;
+import bio.terra.datarepo.model.SnapshotSummaryModel;
+import com.google.cloud.storage.BlobId;
+import common.utils.StorageUtils;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.config.TestUserSpecification;
+import scripts.testscripts.baseclasses.SimpleDataset;
+import scripts.utils.DataRepoUtils;
+
+public class SnapshotScaleDelete extends SimpleDataset {
+  private static final Logger logger = LoggerFactory.getLogger(SnapshotScaleDelete.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public SnapshotScaleDelete() {
+    super();
+  }
+
+  private SnapshotModel snapshotModel;
+
+  private static List<BlobId> scratchFiles = new ArrayList<>();
+  private String dirObjectId;
+
+  private int NUM_SNAPSHOTS = 1;
+
+  public void setParameters(List<String> parameters) {
+    if (parameters != null && parameters.size() > 0) {
+      NUM_SNAPSHOTS = Integer.parseInt(parameters.get(0));
+    }
+    logger.debug("Number of snapshots to create and delete (default is 1): {}", NUM_SNAPSHOTS);
+  }
+
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    // create the profile and dataset
+    super.setup(testUsers);
+
+    // get the ApiClient for the snapshot creator, same as the dataset creator
+    ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
+    RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+
+    // dont bother to load data into the new dataset
+    // create MANY snapshots based on a single dataset
+
+    /*    for (int i = 0; i < NUM_SNAPSHOTS; i++) {
+       // make the create snapshot request and wait for the job to finish
+       // the name of the snapshot will already be randomized based on the bool
+       JobModel createSnapshotJobResponse =
+           DataRepoUtils.createSnapshot(
+               repositoryApi, datasetSummaryModel, "snapshot-simple.json", true);
+
+       // save a reference to the snapshot summary model so we can delete it in cleanup()
+       // TODO ^ shouldn't I be deleting them all anyway?
+       // do I need to account for the fact that many will be deleted already?
+       // another idea would be that we create and then delete them each in the userjourney
+       SnapshotSummaryModel snapshotSummaryModel =
+           DataRepoUtils.expectJobSuccess(
+               repositoryApi, createSnapshotJobResponse, SnapshotSummaryModel.class);
+       logger.info("Successfully created snapshot: {}, index: {}",
+           snapshotSummaryModel.getName(),
+           i
+       );
+    }*/
+  }
+
+  public void userJourney(TestUserSpecification testUser) throws Exception {
```

Comment:
Does this test reproduce the problem? That is, if you remove your fix, does this test fail? It isn't clear to me from looking at this that this would cause the error seen in DR-1703.

Reply:
This test waits for the snapshot creation job to complete before creating another one. Maybe if we created all of the snapshots first and then polled all of the jobs we might see this error.